src/eval/lang.rs

```diff
@@ -127,6 +128,13 @@ impl<F: LurkField, C: Coprocessor<F>> Lang<F, C> {
         self.coprocessors.insert(name, (cproc.into(), z_ptr));
     }
 
+    pub fn add_coprocessor_lem<T: Into<C>, S: Into<Symbol>>(&mut self, name: S, cproc: T) {
+        let name = name.into();
+        // TODO: eliminate this unecessary piece of data
+        let z_ptr = lurk::z_ptr::ZPtr(ExprTag::Nil, F::ZERO);
```

Comment:
It's a little worse than this comment suggests. As far as I can see, this is adding the *wrong* data. Now, maybe in the LEM case, this wrong data is unnecessary. But a `Lang` is not specific to LEM or not-LEM. And consequently, it cannot be correct to construct one this way. This 'scalar ptr' must correspond to the symbol to which the coprocessor is bound. If LEM requires its own variant of `Lang`, then that would need to be formalized somehow.

This is not just academic. A `Lang` created as above could lead a non-LEM circuit to producing incorrect proofs. Such a `Lang` would be toxic and require very special handling. It shouldn't be possible to create such dangerous substances with the Lurk API.

Reply:
The LEM pipeline indeed doesn't need the cached `ZPtr` for this.

Would it be okay to have an `add_coprocessor_with_zptr` and force the caller to provide the `ZPtr`? Then LEM would be able to feed it with whatever.